### PR TITLE
Fixed Z-algo

### PR DESCRIPTION
**Pull request by ongyiumark**
Dumb single character bug that makes it O(n^2)

### CODE REVIEW

**Comment by leloykun** [comment]
@ongyiumark are you sure this works?

**Comment by leloykun** [comment]
I trust you with this haha

**Comment by ongyiumark** [comment]
> @ongyiumark are you sure this works?

Yeah, i just changed `i=0` to `i=1`. It makes sense to ignore `i=0` because the lcp with itself is always the length.

**Review by leloykun — APPROVED** [review]
You own this part of the code now. If anyone asks how to use the Z-algo, imma direct them to you haha

We should add documentation for this in the future, but that's for a different PR

File: pvl/strings/z.hpp [hunk 1]
```diff
@@ -11,7 +11,7 @@ std::vector<int> compute_z(std::string s) {
   int n = s.length();
   std::vector<int> z(n, 0);
   int L = 0, R = 0;
-  for (int i = 0; i < n; i++) {
+  for (int i = 1; i < n; i++) {
```
Comment (leloykun) [inline pvl/strings/z.hpp:14]: add a comment explaining why this must start at `1`

**Review by leloykun — CHANGES_REQUESTED** [review]

**Review by leloykun — APPROVED** [review]

**Pull request merged**